src/controllers/chat.js

```diff
@@ -0,0 +1,58 @@
+const { sign } = require('jsonwebtoken');
+const AllUsers = require('./../model/queries/getAllUsers');
+const getMessages = require('./../model/queries/getAllUserMessages');
+
+// make token for every uesr
+const makeToken = (users) => {
+  const array = users.slice(0, users.length);
+  for (let i = 0; i < array.length; (i += 1)) {
+    const user = { username: array[i].user_name, permission: array[i].permission };
+    array[i] = { username: array[i].user_name, token: sign(user, process.env.SECRET) };
+  }
+  return array;
+};
+
+// Render Chat Page ;
+const renderChat = (req, res) => {
+  let permission = 'doctor';
+  if (res.locals.unlockCookie.permission === permission) {
```

Comment:
if unlockCookie was undefined then this would cause a problem something like cant read permission of undefined